code/modules/mining/minebot.dm

```diff
@@ -312,15 +314,15 @@
 	var/base_speed_add = 1
 	var/base_cooldown_add = 10 //base cooldown isn't reset to normal, it's just added on, since it's not practical to disable the cooldown module
 
-/obj/item/slimepotion/slime/sentience/mining/after_success(mob/living/user, mob/living/simple_animal/SM)
-	if(istype(SM, /mob/living/simple_animal/hostile/mining_drone))
-		var/mob/living/simple_animal/hostile/mining_drone/M = SM
-		M.maxHealth = initial(M.maxHealth) + base_health_add
-		M.melee_damage_lower = initial(M.melee_damage_lower) + base_damage_add
-		M.melee_damage_upper = initial(M.melee_damage_upper) + base_damage_add
-		M.move_to_delay = initial(M.move_to_delay) + base_speed_add
-		if(M.stored_gun)
-			M.stored_gun.overheat_time += base_cooldown_add
+/obj/item/slimepotion/slime/sentience/mining/after_success(mob/living/user, mob/living/simple_animal/simple_mob)
+	if(istype(simple_mob, /mob/living/simple_animal/hostile/mining_drone))
```

Comment:
Early return.

Reply:
Huh? What do you mean? You want me to add an early return?

Reply:
```dm
if (!istype(simple_mob, /mob/living/simple_animal/hostile/mining_drone))
    return

// rest of code
```